### ISSUE TITLE
why test train_accuracy need to set phase_train=True?

### ISSUE THREAD

**Issue by kaizhong1** [rcouturier/steganalysis_with_CNN_and_SRM#1]
The ema have applyed the batch_mean and batch_val when the train_step.run run,why we set train_accuracy=accuracy(phase=True)?  Before the global_test_accuracy ,why we need to compute update of batch normalization?

**Comment by rcouturier** [comment]
In fact we distinguish the training and the testing for batch normalization. Then during the phase train, we will compute the average of ema (with the formula), during the testing we will use the formula computed (during the training). All the codes using batch normalization are based on that. So for globa_test_accuracy (this correspond to the testing part), the boolean value phase_train is set to False

**Comment by kaizhong1** [comment]
Yes,i konw.Phase_train is boolean value to distinguish the training and the testing.And every batch the train_step.run will update the ema,but when calculate train_accuracy,the phase_train=Ture,it will also update the ema.And when we calculate globao_test_accuracy ,why we need to compute update of batch normalization?

**Comment by rcouturier** [comment]
With train_accuracy, maybe it is not necessary to update ema. In fact, we didn't try. I think it will not change significatively the training phase.
For global_test_accuracy or anywhere we need to use batch normalization. In training phase, we need to update ema and in testing phase we simply use it.
